### DISCUSSION TITLE
Replace commons-fileupload with servlet 3.0 multipart support

### DISCUSSION THREAD

**Discussion by erikgb** [eclipse-rdf4j/rdf4j#4370] [Ideas]
I am looking at https://github.com/eclipse/rdf4j/issues/3559, as it represents a total blocker for upgrading spring/spring-boot for us in a project that also has dependencies to rdf4j. This is assumed a significant task, so it has to be divided into parts. And I hope some of it can be done pre-migrating javax -> jakarta.

One blocker I have identified is the dependency we have to commons-fileupload (in rdf4j-http-workbench). commons-fileupload does not have any new releases since 2018, and it should not really be required - as servlet 3.0 added support for in-tree multipart handling: https://docs.oracle.com/javaee/7/tutorial/servlets011.htm

Do you think this is doable? I don't see how I could do this effectively - without some knowledge of how the rdf4j-http-workbench fits in the overall rdf4j ecosystem, and would at least require some input from maintainers!

**Comment by hmottestad** [comment]
I usually use docker to test out my changes in the server or workbench. It's a bit of a slow process though. 

https://github.com/eclipse/rdf4j/blob/main/docker/run.sh

I know it's possible to run your own Servet container locally and connect your IDE so that you can deploy and also debug. But it's been maybe 10 years since the last time I did that. 

If you make a PR with your proposed changes I think I can enable our CI to run all the tests for you to save you from having to do that locally. 

Btw. Are you by any chance working on this for Statnett?

**Reply by erikgb** [reply]
Ok, I'll try to prepare a PR. Do you require an issue as well?

> Btw. Are you by any chance working on this for Statnett?

Yes, it is related to my Statnett engagement, but I plan to contribute to rdf4j as a "hobby" project. 😉

**Reply by erikgb** [reply]
Draft PR here: https://github.com/eclipse/rdf4j/pull/4372 I could mark it as ready, but I suspect it requires some minor adjustments in the multipart-config area. Would you require an issue?

**Reply by abrokenjester** [reply]
I'll convert this discussion to an issue. Easy as.

**Reply by abrokenjester** [reply]
Ah I see you already created an issue : #4374 .

**Reply by erikgb** [reply]
Even a PR: https://github.com/eclipse/rdf4j/pull/4375. 😉 But I suspect it needs some adjustments, and I have no idea how to run the workbench...

**Reply by erikgb** [reply]
I'll need some help testing/verifying the workbench with the proposed changes! I have no idea how the workbench works, and I seem unable to even run (reach) it in my local dev-environment.

This change is a blocker for upgrading beyond Jakarta EE 8.
